jQuery/ajax.js

```diff
@@ -1,82 +1,113 @@
-function ajax(citizens) {
-  var characters = getUniqueValuesForField('character', citizens);
-  var hometowns = getUniqueValuesForField('hometown', citizens);
-  var sizes = getUniqueValuesForField('size', citizens);
-
-  populateSelectWithOptions("#character-select", characters);
-  populateSelectWithOptions("#hometown-select", hometowns);
-  populateSelectWithOptions("#size-select", sizes);
-
-  var selectHandler = function(event) {
-    var params = gatherParams();
-    var filteredCitizens = filter(params, event.data.citizens);
-    redrawContent(filteredCitizens);
+window.hh = {};
+
+window.hh.selectors = {
+  menu: '.menu',
+  content: '.content',
+  nameInput: '#name-input',
+  characterSelect: '#character-select',
+  hometownSelect: '#hometown-select',
+  sizeSelect: '#size-select'
+};
+
+/* Defines functions that interact directly with the DOM */
+window.hh.domManipulation = window.hh.domManipulation || (function() {
+  function populateSelectWithOptions(select, options) {
```

Comment:
This function seems needlessly specific. You're taking one element and populating it with other elements. You see where I'm going with this?

consider a world where we have a pure function that accepts two parameters, a parent and a child:

```js
function _append(parent, child) {
  // Both are wrapped in jQuery objects just in case they were
  // passed in as native DOM elements. 
  $(parent).append($(child));
}
```

And consider that the other thing you're doing is creating new html elements every time based on some string (html or form field values)

```js
function _create(type, options) {
  return $(type, options); 
}
```

And now you can use this function like so!

(I'm renaming it `populate` because I'm a crochety asshole

```js
function populate(element, children) {
 children.forEach(function(child) {
    _append( select, _create("<option>", { value: child, html: child } )
  });
}
```

If I wanted to be even more pretentious I might just use a pure function for each lol

```js
function _each(collection, fn) {
  collection.forEach(fn);
}
```

aaaand boom

```js
function populate(element, children) {
 _each(children, function(child) {
    _append( select, _create("<option>", { value: child, html: child } )
  });
}
```

Kinda more generic this way. I think I had more fun coding it than actually making it useful. oops.